sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/AbstractLNAdapter.java

```diff
@@ -980,6 +998,23 @@ public ControlBlockAdapter createControlBlockIfNotExists(String cbName, String i
                 );
     }
 
+    /**
+     * Generate a ControlBlockId based on the current LN and the given ldName (ldName can be different from the parent LD.name)
+     *
+     * @param ldName LD name to use for generating the id
+     * @param cbName name of the ControlBlock
+     * @return "ldName/LnPrefixLnClassLnInst.cbName". Blank values are omitted (e.g "IEDNAME1LD1/LLN0.CBNAME1")
+     */
+    public String generateControlBlockId(String ldName, String cbName) {
+        return Utils.emptyIfBlank(ldName)
```

Comment:
StringUtils is enough :  
```java
return StringUtils.trimToEmpty(ldName)
                + "/"
                + StringUtils.trimToEmpty(getPrefix())
                + StringUtils.defaultString(getLNClass())
                + StringUtils.trimToEmpty(getLNInst())
                + "."
                + StringUtils.trimToEmpty(cbName);
```

Reply:
OK. I did not know about `trimToEmpty`. I delete Utils.emptyIfBlank method and replace it everywhere.
And I also use `defaultString`